### ISSUE TITLE
Drag and Drop not working

### ISSUE THREAD

**Issue by AkyrosXD** [0xd4d/dnSpy#944]
I'm using dnspy v5.0.7 and I'm having a problem while trying to drop a .NET assembly or exe in Assembly Explorer. I can just use File > Open but for me Drag and drop is a faster way and it doesn't work for me when I try to drop a file in the assembly explorer.

**Comment by TheProxyRE** [comment]
Its probably because you run it as administrator. (This is windows problem and when application is executed as administrator it doesn't allow drag and drop)

**Comment by AkyrosXD** [comment]
> Its probably because you run it as administrator. (This is windows problem and when application is executed as administrator it doesn't allow drag and drop)

I never run it as Administrator and still won't let me use drag and drop. The previous version of dnspy that I had (v5.0.2 I think) it worked fine but when I updated dnspy the drag and drop stopped working.

**Comment by 0xd4d** [comment]
It depends on what you're dragging and dropping. You can disable opening unsupported files in the settings, perhaps you enabled it.

**Comment by AkyrosXD** [comment]
> It depends on what you're dragging and dropping. You can disable opening unsupported files in the settings, perhaps you enabled it.

I checked the settings and I have it on "All" and even if I try to drop .NET files it still won't work.

**Comment by gagmeng** [comment]
This problem also exists on my computer, but it's fine in dnspy 4.5.x.

**Comment by r3xq1** [comment]
I have the same problem in version 5.0 But you can integrate the launch of the program into the context menu and from there run the necessary program for decompiling

**Comment by piratesephiroth** [comment]
I don't have this problem. Are you using the appveyor build?

**Comment by 0xd4d** [comment]
Post some more info, because it's impossible for us to figure out why it won't work for you. Works for me (TM).

OS? Buggy antivirus? etc.

**Comment by r3xq1** [comment]
dnSpy v5.0.7 (x64. Administator).  - Tried old versions
Windows 7 x64 Max
no Anti-Virus

**Comment by r3xq1** [comment]
> You can't drag and drop if you start it as admin, see @TheProxyRE 's reply above
I started even without the admin. Did not help(

**Comment by 0xd4d** [comment]
It's caused by this PR: https://github.com/0xd4d/dnSpy/pull/904 /cc @LFriede

The cast fails on Windows 7.

**Comment by AkyrosXD** [comment]
> Post some more info, because it's impossible for us to figure out why it won't work for you. Works for me (TM).
> 
> dnSpy version? OS version? Buggy antivirus? etc.

dnSpy version: v5.0.7
OS Version: Windows 7 Professional
Anti-Virus: still the same thing even if its disabled

**Comment by LFriede** [comment]
> 
> 
> It's caused by this PR: #904 /cc @LFriede
> 
> The cast fails on Windows 7.

Damn, sorry! I'll take a look at it some time soon.

**Comment by LFriede** [comment]
It looks like the way I resolved the links has problems with UAC, even if disabled. I tried another method in my fork which seems to work fine. Would be nice if the people who encountered problems can verify it.
https://github.com/LFriede/dnSpy
I've attached binaries in the release page of my fork too.

**Comment by 0xd4d** [comment]
It's better to send a PR.

**Comment by 0xd4d** [comment]
Latest build has the fix: https://ci.appveyor.com/project/0xd4d/dnspy/branch/master/artifacts

**Comment by LFriede** [comment]
```csharp
var directory = Path.GetDirectoryName(filenames[i]);
var file = Path.GetFileName(filenames[i]);

var shell = new Shell32.Shell();
var folder = shell.NameSpace(directory);
var folderItem = folder.ParseName(file);
if (folderItem != null && folderItem.IsLink) {
	var link = (Shell32.ShellLinkObject)folderItem.GetLink;
	filenames[i] = link.Path;
}
```

This was the old code. `folderItem.GetLink` threw exceptions for example if a desktop link was stored for all users and not in the users desktop directory. I'm not sure but I would guess that the class opens them internally with write access or something like that. So I didn't saw a solution to get it working like this and replaced the code with a snippet using other classes to do the same thing and it worked. I tested on another Win7 box with UAC disabled, drag and drop was not working at all on this one, I did no debugging on this machine because I was to lazy to install Visual Studio on it. The alternative code worked on that machine too so I think it's fixed now.
